pandas/_typing.py

```diff
@@ -216,6 +216,7 @@
 # Using List[int] here rather than Sequence[int] to disallow tuples.
 ScalarIndexer = Union[int, np.integer]
 SequenceIndexer = Union[slice, List[int], np.ndarray]
+TakeIndexer = Union[Sequence[int], np.ndarray]
```

Comment:
```suggestion
TakeIndexer = Union[Sequence[ScalarIndexer], npt.NDArray[np.integer]]
```

I don't think that allowing Sequence[np.integer] breaks the EA api contract. Allowing numpy ints where we specify int is probably also a given.

(OT we should probably rename ScalarIndexer to IntLike (or preferably _IntLike_co for consistency with numpy after some consideration of allowing bool)

Reply:
Thank you! I committed your change and rebased.

Reply:
Probably would be better to use:
```py
TakeIndexer = Union[Sequence[int], Sequence[np.integer], npt.NDArray[np.integer]]
```
to not allow a sequence of mixed types.